### PR TITLE
Generate all valid parentheses (Backtracking Algorithm) #2926

### PR DESCRIPTION
**Pull request by kanchan1910**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Program to generate all combinations of well-formed parentheses.

### Checklist

- [ yes] I've read the contribution guidelines.
- [ yes] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

### CODE REVIEW

**Comment by kanchan1910** [comment]
@bhav09, Kindly review this, and if any changes required please let me know.

Thank you sir for your time!

File: C-Plus-Plus/cp/Generate_all_valid_parentheses_combinations.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
+/*
+Problem statement : Given n pairs of parentheses, we have to write a function to generate all combinations of well-formed parentheses.
+
+Sample test cases :
```
Comment (bhav09) [inline C-Plus-Plus/cp/Generate_all_valid_parentheses_combinations.cpp]: Can you please add the test case at the end of the code after the complexity.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

**Comment by kanchan1910** [comment]
@bhav09 sir, kindly check now!

File: C-Plus-Plus/cp/Generate_all_valid_parentheses_combinations.cpp [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+int main()
+{
+	int n;
+
+	cout << "Enter the number:";
+
+	cin >> n;
+
+	vector<string>ans;  // this is our output vector
```
Comment (bhav09) [inline C-Plus-Plus/cp/Generate_all_valid_parentheses_combinations.cpp]: Comments should be above the line of code. and not alongside. Please work on it. Rest everything is fine. Great work!

**Review by bhav09 — CHANGES_REQUESTED** [review]
Add all the comments should be above the line of code. Rest, perfect.

**Comment by bhav09** [comment]
Great work @kanchan1910!

**Comment by kanchan1910** [comment]
@bhav09 Thank you so much, sir!

**Comment by kanchan1910** [comment]
@bhav09, sir, kindly check this, why my score is not getting updated.

https://gssoc.girlscript.tech/profile.html?id=kanchan1910

Thank you for your time

**Pull request merged**